test/apps/iodemo/io_demo.cc

```diff
@@ -1332,7 +1337,8 @@ class DemoClient : public P2pDemoCommon {
             _client->handle_operation_completion(_server_index, IO_READ,
                                                  _iov->data_size());
 
-            if (_validate && (_status == UCS_OK)) {
+            if (_validate && (_status == UCS_OK) &&
+                (server_info.conn->ucx_status() == UCS_OK)) {
```

Comment:
would not `_status` check be enough here? Can it be `UCS_OK`, while the data is invalid?

Reply:
yes, makes sense to me, if the operation completed with `UCS_OK`, validation must be ok.
done